packages/contracts-bedrock/src/dispute/AnchorStateRegistry.sol

```diff
@@ -23,105 +24,192 @@ import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
+    /// @custom:legacy
+    /// @notice Returns the anchor root. Note that this is a legacy deprecated function and will
+    ///         be removed in a future release. Use getAnchorRoot() instead. Anchor roots are no
+    ///         longer stored per game type, so this function will return the same root for all
+    ///         game types.
+    function anchors(GameType /* unused */ ) external view returns (Hash, uint256) {
+        return getAnchorRoot();
+    }
+
+    /// @notice Returns the current anchor root.
+    /// @return The anchor root.
+    function getAnchorRoot() public view returns (Hash, uint256) {
+        // Return the starting anchor root if there is no anchor game.
+        if (address(anchorGame) == address(0)) {
+            return (startingAnchorRoot.root, startingAnchorRoot.l2BlockNumber);
+        }
+
+        // Otherwise, return the anchor root.
+        return (Hash.wrap(anchorGame.rootClaim().raw()), anchorGame.l2BlockNumber());
+    }
+
+    /// @notice Determines whether a game is registered in the DisputeGameFactory.
+    /// @param _game The game to check.
+    /// @return Whether the game is factory registered.
+    function isGameRegistered(IDisputeGame _game) public view returns (bool) {
+        // Grab the game and game data.
+        (GameType gameType, Claim rootClaim, bytes memory extraData) = _game.gameData();
+
+        // Grab the verified address of the game based on the game data.
+        (IDisputeGame _factoryRegisteredGame,) =
+            disputeGameFactory.games({ _gameType: gameType, _rootClaim: rootClaim, _extraData: extraData });
+
+        // Return whether the game is factory registered.
+        return address(_factoryRegisteredGame) == address(_game);
+    }
+
+    /// @notice Determines whether a game is of a respected game type.
+    /// @param _game The game to check.
+    /// @return Whether the game is of a respected game type.
+    function isGameRespected(IDisputeGame _game) public view returns (bool) {
+        return _game.gameType().raw() == portal.respectedGameType().raw();
+    }
+
+    /// @notice Determines whether a game is blacklisted.
+    /// @param _game The game to check.
+    /// @return Whether the game is blacklisted.
+    function isGameBlacklisted(IDisputeGame _game) public view returns (bool) {
+        return portal.disputeGameBlacklist(_game);
+    }
+
+    /// @notice Determines whether a game is retired.
+    /// @param _game The game to check.
+    /// @return Whether the game is retired.
+    function isGameRetired(IDisputeGame _game) public view returns (bool) {
+        return _game.createdAt().raw() < portal.respectedGameTypeUpdatedAt();
+    }
+
+    /// @notice Determines whether a game resolved properly and the game was not subject to any
+    ///         invalidation conditions. The root claim of a proper game IS NOT guaranteed to be
+    ///         valid. The root claim of a proper game CAN BE incorrect and still be a proper game.
```

Comment:
Nice to have this documented, not very intuitive. Thanks for getting that in, must yell this at third-party consumers of this contract...